docs/pages/database-access/guides/azure-sql-server-ad.mdx

```diff
@@ -125,30 +126,106 @@ $ teleport db configure create \
    -o file \
    --token=/tmp/token \
    --proxy=mytenant.teleport.sh:443 \
-   --name=sqlserver \
-   --protocol=sqlserver \
-   --uri=my-server.database.windows.net:1433 \
-   --labels=env=dev
+   --azure-sqlserver-discovery=eastus
 ```
 </ScopedBlock>
 
+The command will generate a Database Service configuration with Azure SQL
+Server auto-discovery enabled in the `eastus` region and place it at the
+`/etc/teleport.yaml` location.
+
+## Step 7/9. Configure IAM permissions for Teleport
+
+The Teleport Database Service needs Azure IAM permissions to discover and
+register SQL Server instances.
+
+### Configure an Azure service principal
+
+(!docs/pages/includes/database-access/azure-configure-service-principal.mdx!)
+
+### Create a custom role
+
+Teleport requires `<resource-type>/read` permissions for discovery, but
+Teleport only needs permissions for the resource types you have.
+
+Here is a sample role definition allowing Teleport to read **Azure SQL Servers**
+and **Azure SQL Managed Instances**:
+
+```json
+{
+    "properties": {
+        "roleName": "SQLServerAutoDiscovery",
+        "description": "Allows Teleport to discover SQL Servers and SQL Managed Instances.",
+        "assignableScopes": [
+            "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+        ],
+        "permissions": [
+            {
+                "actions": [
+                    "Microsoft.Sql/managedInstances/read",
+                    "Microsoft.Sql/servers/read"
+                ],
+                "notActions": [],
+                "dataActions": [],
+                "notDataActions": []
+            }
+        ]
+    }
+}
+```
+
 <Admonition
   type="tip"
   title="Tip"
 >
-  A single Teleport process can run multiple different services, for example
-  multiple Database Service agents as well as the SSH Service or Application
-  Service.
-  
+  The `assignableScopes` field above includes a subscription
+  `/subscriptions/<subscription>`, allowing the role to be assigned at any
+  resource scope within that subscription or the subscription scope itself.
+  If you want to further limit the `assignableScopes`, you can use a resource
+  group `/subscriptions/<subscription>/resourceGroups/<group>` or a management
+  group `/providers/Microsoft.Management/managementGroups/<group>` instead.
 </Admonition>
 
+Now go to the [Subscriptions](https://portal.azure.com/#view/Microsoft_Azure_Billing/SubscriptionsBlade)
+page and select a subscription.
+
+Click on **Access control (IAM)** in the subscription and select **Add** >
+**Add custom role**:
+
+<Figure align="left">
```

Comment:
These images are large enough that they fill the entire space of the content pane in our docs, so the use of the Figure component to align them is superfluous. It's not doing any harm, but you could remove them if you want.